### ISSUE TITLE
Combine call.Param list actions

### ISSUE THREAD

**Issue by Jaymon** [Jaymon/endpoints#138]
Right now there are three different list actions:

* "store_list" - set to have a value like `1,2,3` be blown up to `['1', '2', '3']`
* "append" - if multiple param values should be turned into an array (eg, `foo=1&foo=2` would become `foo=[1, 2]`)
* "append_list" - it's store_list + append, so `foo=1&foo=2,3` would become `foo=[1, 2, 3]`

I'm sure I had a reason to have all these be separate at some point in the past but the reasoning has been completely lost to time and I think `append_list` should be the only way.

So basically, remove `append_list` and `append` and take the functionality of the current `append_list` and rename it `store_list`.

**Comment by Jaymon** [comment]
[argparse action](https://docs.python.org/3/library/argparse.html#action) has `append` and `extend`, I think I can merge `store_list` and `append_list` into `extend` and keep `action` as is. I think that would mostly be consistent with how argparse does it.
